### PR TITLE
#175636852 Add User Role-Setting for Super Administrator to Assign Roles to Users Using Their Emails 

### PR DESCRIPTION
**Pull request by Thiernope**
#### What does this PR do?
Add User Role-Setting for Super Administrator to Assign Roles to Users Using Their Emails
#### Description of Task to be completed?
- create userRoles models 
- add column of roleId in the User model
- create superAdmin role
- Add functionalities to create, read, update and delete users' roles using their emails
- Restrict the deletion of SUPER_ADMIN role in the system

#### How should this be manually tested?

- run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd directory tech-strikers-bn-backend`
- run `git checkout ft-user-role-setting-175636852`
- run `npm install`
- run`npm run role:create`
- run`npm run admin:create`
- add admin password in your .env file e.g `SUPER_ADMIN_PASSWORD = superAdmin`
-user the admin email that is created to login and get token for authorization
- Referring to the screenshots below, use postman, create a role in the system
- assign that role to a user in the database using the his/her email
- get all roles available in the system 
- delete any role using its id
- upadtate any role available in the system

#### Any background context you want to provide?
admin is created once  his role is already created. `npm run role:create` must be executed first in order to execute `npm run admin:create`
#### What are the relevant pivotal tracker stories?
[#175636852](https://www.pivotaltracker.com/n/projects/2475704/stories/175636852)
#### Screenshots (if appropriate)
Create role
create role
Assign role 
assign role
Get all roles available in the system
get all roles
Get one specific role
get one role
Update one specific role using its id
update role
Delete one specific role using its id
delete role

#### Questions:
No so far

### CODE REVIEW

File: src/database/migrations/20201214101348-create-user-roles.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable('userRoles');
+  }
+};
```
Comment (fabriceumuhire) [inline src/database/migrations/20201214101348-create-user-roles.js:30]: Add an extra line

File: src/routes/api/users.route.js [hunk 2]
```diff
@@ -20,6 +25,25 @@ router.get("/user", tokenAuth, getUserInfo);
...
+router.get("/user", (req, res) => {
+  res.status(200).json({ message: "successfully sent" });
+});
+
+router.put("/user", (req, res) => {
+  res.status(200).json({ message: "successfully sent" });
+});
+
+router.post("/user/signup", checkEmailExist, signup);
+router.post("/user/assigRole", isSuperAdmin, roleAssignValidation, assign);
```
Comment (fabriceumuhire) [inline src/routes/api/users.route.js]: Can you update to `assignRole` instead of `assigRole`?

**Comment by fabriceumuhire** [comment]
Great work, looks good for me. Just add extra lines to some files and correct some typos.

File: src/database/migrations/20201214101348-create-user-roles.js (hunk 1 above)
Comment (Thiernope) [reply src/database/migrations/20201214101348-create-user-roles.js:30]: I see that there is no red sign. The last line is there. unless I want to put more line at the end, and this is not accepted.

**Review by Muhire-Josue — APPROVED** [review]
LGTM

**Pull request merged**